src/lib/config-win32.h

```diff
@@ -190,6 +190,9 @@
 /* Define if you have the sendto function. */
 #define HAVE_SENDTO 1
 
+/* Define if you have the strnlen function. */
+#define HAVE_STRNLEN 1
+
```

Comment:
Well, I guess we need a
```
if !defined(__WATCOMC__)
```
around this since their library doesn't have strnlen like MSVC and MinGW do...

Reply:
> Glad we have all these CI systems building c-ares to catch this stuff.

Yes it is good because I'm not very careful with this PR. I can add the `strnlen_s()` variant for Watcom tomorrrow. The code already does a NULL check and returns 0 so that aligns with the `strnlen_s()` behavior.

Reply:
It appears that in order to access `strnlen_s()` in Watcom you need to define `__STDC_WANT_LIB_EXT1__` before including `<string.h>`. Do you want that?

Reply:
I could go either way on that.  Either just use the fallback strnlen code as performance doesn't matter for something like openwatcom, or just turn that flag on at the top of this header.  honestly, not sure when this was introduced to openwatcom, if someones using an older version than 2.0 (like 1.9) it might not even exist there.

Reply:
I'm seeing that `strnlen_s()` is in C17 in (normative) Annex K.3.7.4.4. Any implementation that defines `__STDC_LIB_EXT1__` will have `strnlen_s()`, and they will only exist if `__STDC_WANT_LIB_EXT1__` is set to nonzero by the programmer. I guess that instead of the Watcom macro we can just check for these.